### DISCUSSION TITLE
Deleting database

### DISCUSSION THREAD

**Discussion by JHarrison712** [netbox-community/netbox#12213] [Help Wanted!]
I recently cloned my Netbox instance for a Dev and Prod environment. I want to perform a database delete so it deletes all the dev entries ( Kinda like a fresh installation ) but I can't find any documentation that has the steps on it? Has anyone done with before or is it best to just build a new VM again and do it from scratch?

**Comment by candlerb** [comment]
Drop the database, [recreate the database](https://docs.netbox.dev/en/stable/installation/1-postgresql/#database-creation) (empty), then [run the upgrade script](https://docs.netbox.dev/en/stable/installation/3-netbox/#run-the-upgrade-script) to recreate all the tables.

These steps are basically the same as [loading an exported database](https://docs.netbox.dev/en/stable/administration/replicating-netbox/#load-an-exported-database), except that you run `./upgrade.sh` instead of loading the database backup.

**Reply by JHarrison712** [reply]
Hmm maybe I didn't do something right. I did the following 

- sudo -u postgres psql
- - drop database netbox
- - delete database netbox
- - create database netbox
- - CREATE USER netbox WITH PASSWORD 'secretkeyhere'
- - GRANT ALL PRIVILEGES ON DATABASE netbox TO netbox
- \q
- sudo /opt/netbox/upgrade.sh
- sudo systemctl restart netbox
- sudo systemctl restart netbox-rq

and I still have the old data https://i.imgur.com/9P73Mwc.png

**Reply by candlerb** [reply]
Then presumably you dropped the wrong database.

Look at `/opt/netbox/netbox/netbox/configuration.py` to see what database it's pointing to - which postgres server, and which database on that server.

Or else, the commands didn't complete properly (did you see any errors in response?)

**Reply by JHarrison712** [reply]
I think I figured it out I was missing the ; at the very end of everything.

**Reply by JHarrison712** [reply]
sudo -u postgres psql
\l - see if netbox is in the database
drop database netbox;
\l - do you still see netbox?
create database netbox;
\1 - now you see the new database for netbox
CREATE USER netbox WITH PASSWORD 'secretkeyhere';
GRANT ALL PRIVILEGES ON DATABASE netbox TO netbox;
\q
sudo /opt/netbox/upgrade.sh
source /opt/netbox/venv/bin/activate
python3 /opt/netbox/netbox/manage.py createsuperuser
sudo systemctl restart netbox
sudo systemctl restart netbox-rq
